libcaf_io/test/basp_udp.cpp

```diff
@@ -0,0 +1,1094 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright (C) 2011 - 2017                                                  *
+ * Dominik Charousset <dominik.charousset (at) haw-hamburg.de>                *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#include "caf/config.hpp"
+
+#define CAF_SUITE io_basp_udp
+#include "caf/test/dsl.hpp"
+
+#include <array>
+#include <mutex>
+#include <memory>
+#include <limits>
+#include <vector>
+#include <iostream>
+#include <condition_variable>
+
+#include "caf/all.hpp"
+#include "caf/io/all.hpp"
+
+#include "caf/io/dgram_handle.hpp"
+#include "caf/io/dgram_servant.hpp"
+
+#include "caf/deep_to_string.hpp"
+
+#include "caf/io/network/interfaces.hpp"
+#include "caf/io/network/ip_endpoint.hpp"
+#include "caf/io/network/test_multiplexer.hpp"
+
+namespace {
+
+struct anything { };
+
+anything any_vals;
+
+template <class T>
+using maybe = caf::variant<anything, T>;
```

Comment:
I think this file contains much duplicated code from `basp.cpp` right? We might can unify that?

Reply:
The fixtures could be unified, but I think it's ok for now.

Reply:
> The fixtures could be unified, but I think it's ok for now.

Yes, better than putting everything into a single huge test suite.

However, that reminds me that there exists already some I/O specific boilerplate code in https://github.com/actor-framework/actor-framework/blob/master/libcaf_test/caf/test/io_dsl.hpp. Seems worthwhile to leverage this, but I don't know how much effort it is. For now, I think copy & paste is fine but we should address this eventually.